### PR TITLE
Added the GetColumn functionality to dynamic API

### PR DESCRIPTION
**Pull request by Zruty0**
Improves #985 .
Adds a `GetColumn<T>`("columnName")` functionality to eagerly extract a column from a data view

### CODE REVIEW

File: src/Microsoft.ML.Api/ColumnCursor.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Api
+{
+    /// <summary>
+    /// Extension metrods that allow to extract values of a single column of an <see cref="IDataView"/> as an
```
Comment (sfilipi) [inline src/Microsoft.ML.Api/ColumnCursor.cs]: >metrods [](start = 18, length = 7)

typo

File: src/Microsoft.ML.Api/ColumnCursor.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    public static class ColumnCursorExtensions
+    {
+        /// <summary>
+        /// Extract all values of one column of the data view in a form of an <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the values. This must match the actual column type.</typeparam>
+        /// <param name="data">The data view to get the column from.</param>
+        /// <param name="env">The current host environment.</param>
+        /// <param name="columnName">The name of the column to extract.</param>
+        /// <returns></returns>
```
Comment (sfilipi) [inline src/Microsoft.ML.Api/ColumnCursor.cs]: ><returns></returns [](start = 12, length = 18)

fill

File: test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    public sealed class GetColumnTests : BaseTestClass
+    {
+        public GetColumnTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void TestGetColumn()
+        {
+            var path = GetDataPath("breast-cancer.txt");
```
Comment (sfilipi) [inline test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs]: >"breast-cancer.txt" [](start = 35, length = 19)

use TestDatasets

File: test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    Exception e;
+                    for (e = ex; e.InnerException != null; e = e.InnerException)
+                    {
+                    }
+                    Assert.True(e is ArgumentOutOfRangeException || e is InvalidOperationException);
+                    Assert.True(e.IsMarked());
+                }
+            };
+
+            var enum1 = data.AsDynamic.GetColumn<float>(env, "floatScalar").ToArray();
```
Comment (sfilipi) [inline test/Microsoft.ML.Tests/Scenarios/GetColumnTests.cs]: >floatScalar [](start = 62, length = 11)

is there a way to avoid hardcoding?

**Comment by Zruty0** [comment]
Improves #985 .
Adds a `GetColumn<T>("columnName")` functionality to eagerly extract a column from a data view

**Pull request merged**